src/vocab.rs

```diff
@@ -153,6 +153,24 @@ impl SubwordVocab {
             Some(indices)
         }
     }
+
+    fn ngrams_indices(&self, word: &str) -> Option<Vec<(String, usize)>> {
```

Comment:
Can also be `Vec<(&str, usize)>`.

Is this made public somewhere? Also, since this is non-essential functionality, but for very specific use cases, we may want to put this method in a separate trait that users could import when they need this functionality.